### PR TITLE
Chronos integration

### PR DESCRIPTION
**Pull request by thorondir**

### CODE REVIEW

File: database/102-up-timescale/20230508182333-winners_add_application_chronos.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+-- migrate:up
+
+ALTER TYPE ethereum_application ADD VALUE IF NOT EXISTS 'chronos';
```
Comment (af-afk) [inline database/102-up-timescale/20230508182333-winners_add_application_chronos.sql:4]: Don't have "IF NOT EXISTS" - just add it as-is

File: database/102-up-timescale/20230508182333-winners_add_application_chronos.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+-- migrate:down
+-- nothing here
```
Comment (af-afk) [inline database/102-up-timescale/20230508182333-winners_add_application_chronos.sql:7]: Should remove the type here

File: common/ethereum/applications/chronos/chronos.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+)
+
+// GetChronosFees returns Chronos' fee based on fees given by Chronos' pair factory
+// If the token swapped from was the fluid token, get the exact amount,
+// otherwise approximate the cost based on the received amount of the fluid token
+func GetChronosFees(transfer worker.EthereumApplicationTransfer, client *ethclient.Client, fluidTokenContract ethCommon.Address, tokenDecimals int) (*big.Rat, error) {
+	// decode the amount of each token in the log
+	// doesn't contain addresses, as they're indexed
+	if len(transfer.Log.Topics) < 1 {
+		return nil, fmt.Errorf("No log topics passed!")
```
Comment (af-afk) [inline common/ethereum/applications/chronos/chronos.go:147]: Better to return the errors beginning with lowercase as per Go style

File: common/ethereum/applications/chronos/chronos.go [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+	feeNumerator, err := ethereum.CoerceBoundContractResultsToInt(feeNumerator_)
+
+	if err != nil {
+		return nil, fmt.Errorf(
+			"Failed to coerce pair fee numerator! %v",
+			err,
+		)
+	}
+
+	feeMultiplier = new(big.Rat).SetFrac(feeNumerator, big.NewInt(FeeDenominator))
```
Comment (af-afk) [inline common/ethereum/applications/chronos/chronos.go:322]: What's this for?

**Review by af-afk — CHANGES_REQUESTED** [review]

File: common/ethereum/applications/chronos/chronos.go (hunk 4 above)
Comment (thorondir) [reply common/ethereum/applications/chronos/chronos.go:322]: getting the fee% with correct numerator/denominator

**Review by af-afk — APPROVED** [review]

**Pull request merged**